components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -163,6 +164,11 @@ public void updateConfiguration(String tenantDomain, Map<String, String> configu
                         configProperties[j] = property;
                         configProperties[j].setDisplayName(propertyFriendlyNames.get(configProperties[j].getName()));
                         configProperties[j].setDescription(propertyDescriptions.get(configProperties[j].getName()));
+                        if (confidentialProperties != null && confidentialProperties.get(configProperties[j].getName())
+                                != null) {
+                            configProperties[j].setConfidential(
+                                    Boolean.parseBoolean(confidentialProperties.get(configProperties[j].getName())));
```

Comment:
why do we need to use Boolean.parseBoolean? 

What do we return in getConfidentialPropertyValues method?

Reply:
Change the return type of getConfidentialPropertyValues to Map<String, Boolean) from Map<String, String)